src/DtoAbstract.php

```diff
@@ -142,59 +96,79 @@ public function validate(): Validator
     {
-        $this->setValidationFlag(false);
+        /** @var DtoAbstract $dto */
+        $dto = (new static)->populateFromModel($model);
 
-        return $this;
-    }
+        $dto->isValidation() && $dto->validate();
 
-    /**
-     * @param bool $flag
-     */
-    protected function setValidationFlag(bool $flag)
-    {
-        $this->validation = $flag;
+        return $dto;
     }
 
     /**
-     * @param array $data
+     * @param Model[]|Collection $models
      *
-     * @return bool
+     * @return Collection|$self[]
      */
-    protected function isArrayOfSubtypes($key): bool
+    public static function fromModels(Collection $models): Collection
     {
-        // will not work if dto gets updated (dtos are only meant to be populated once)
-        return is_array($this->items[$key]);
+        return $models->map(function (Model $model) {
+            return static::fromModel($model);
+        });
     }
 
     /**
-     * @param string $subtype
-     * @param array  $data
+     * Populate DTO and sub-DTO-s from data array
+     *
+     * - Can be used in controllers, to fill from Request::all()
+     * - Only sets values for keys that are predefined in DTO
+     *
+     * @param array $data
      *
-     * @return array
+     * @return $this
+     * @throws \Exception
      */
-    protected function createSubtype(string $subtype, array $data): array
+    public static function fromArray(array $data): DtoAbstract
     {
-        $dto = new $subtype;
-        $dto->setValidationFlag($this->validation);
+        $dto = new static;
 
-        return $dto->populateFromArray($data)->toArray();
+        collect(array_intersect_key($data, $dto->toArray()))
+            ->filter(function ($value) {
+                return $value !== null && !empty($value);
+            })
+            ->map(function ($value, $key) use ($dto) {
+                if (is_array($value) && $dto->getSubtype($key)) {
+                    $value = $dto->getSubtype($key)::from($value)->toArray();
+                }
+
+                $dto[$key] = $value;
+            });
+
+        $dto->isValidation() && $dto->validate();
+
+        return $dto;
     }
 
     /**
-     * @param Model[]|Collection $models
+     * @param Model|Collection|Model[]|array $source
      *
-     * @return Collection|DtoAbstract[]
+     * @return $this|Collection|$this[]
      */
-    public static function populateFromModels(Collection $models): Collection
+    public static function from($source)
     {
-        return $models->map(function (Model $model) {
-            return (new static)->populateFromModel($model);
-        });
+        if ($source instanceof Collection) {
+            return static::fromModels($source);
+        } elseif ($source instanceof Model) {
+            return static::fromModel($source);
+        } elseif (is_array($source)) {
+            return static::fromArray($source);
+        }
+
+        return collect();
```

Comment:
why return an empty collection here? feels a bit unexpected to me

Reply:
@lilyrae added description about the `collect()` to PHPDoc of method.